src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeSummaryMapper.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.blobrouter.data;
+
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.data.model.EnvelopeSummary;
+import uk.gov.hmcts.reform.blobrouter.data.model.Event;
+import uk.gov.hmcts.reform.blobrouter.data.model.Status;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@Component
+public class EnvelopeSummaryMapper implements RowMapper<EnvelopeSummary> {
+
+    @Override
+    public EnvelopeSummary mapRow(ResultSet rs, int rowNum) throws SQLException {
+        return new EnvelopeSummary(
+            rs.getString("container"),
+            rs.getString("file_name"),
+            rs.getTimestamp("file_created_at").toInstant(),
+            rs.getTimestamp("file_dispatched_at") == null
+                ? null : rs.getTimestamp("file_dispatched_at").toInstant(),
+            Status.valueOf(rs.getString("status")),
+            rs.getBoolean("is_deleted"),
+            rs.getString("event") == null ? null : Event.valueOf(rs.getString("event")),
+            rs.getString("notes"),
+            rs.getTimestamp("event_created_at") == null
```

Comment:
event creation is never null. notes though are. unless `getString` handles it there's nothing else to do :)

Reply:
only if there are events for the envelope, otherwise it is null

Reply:
I think the envelope is created together with an event, in a single transaction. Not sure the mapper needs to be aware of it, though.

Reply:
well. event i always present. whereas envelope is created at the certain condition. since there's no link, should just trust on events first

Reply:
I suggest bullet proof approach